gpMgmt/bin/gppylib/commands/base.py

```
@@ -662,7 +662,11 @@ def findCmdInPath(cmd):
                 CMD_CACHE[cmd] = f
                 return f
 
-        logger.critical('Command %s not found' % cmd)
+        # missing_ok is just suppressing the critical log message. This function
+        # still raises an exception and leaves the client to decide whether to
+        # consume the error.
+        if not missing_ok:
+            logger.critical('Command %s not found' % cmd)
```

Don't we need to print a warning here? The caller just ignores the exception, `package.py:32`

---

It has been logged from the caller. I don't think adding an additional warning message is helpful. What do you think about it?

```
20231117:15:01:01:291021 gpinitstandby:laptop:v-[INFO]:-Syncing Greenplum Database extensions to stand
by
20231117:15:01:01:291021 gpinitstandby:laptop:v-[INFO]:-did not find gppkg binary. skip package sync
2
```